src/inspect_ai/_util/local_server.py

```diff
@@ -0,0 +1,357 @@
+import json
+import logging
+import os
+import random
+import socket
+import subprocess
+import time
+from typing import Any, Dict, Optional, Tuple
+
+import requests
```

Comment:
What would you think about using `httpx` here instead? (since we already depend on it and it works w/ anyio)